dev/io.openliberty.mcp.internal_fat/fat/src/io/openliberty/mcp/internal/fat/tool/ToolErrorHandlingTest.java

```diff
@@ -0,0 +1,340 @@
+                              }
+                            ]
+                          }
+                        }
+                        """;
+
+        JSONAssert.assertEquals(expectedResponse, response, true);
+    }
+
+    @Test
+    public void testToolThrowsExceptionWrappedBySuperclass() throws Exception {
+        String request = """
+                        {
+                          "jsonrpc": "2.0",
+                          "id": 1,
+                          "method": "tools/call",
+                          "params": {
+                            "name": "superclassWrappedExceptionTool",
+                            "arguments": {
+                              "input": "test arg"
+                            }
+                          }
+                        }
+                        """;
+
+        String response = HttpTestUtils.callMCP(server, ENDPOINT, request);
+
+        String expectedResponse = """
+                        {
+                          "id": 1,
+                          "jsonrpc": "2.0",
+                          "result": {
+                            "isError": true,
+                            "content": [
+                              {
+                                "type": "text",
+                                "text": "Invalid input for superclass: test arg"
+                              }
+                            ]
+                          }
+                        }
+                        """;
+        JSONAssert.assertEquals(expectedResponse, response, true);
+    }
+
+    @Test
+    public void testToolThrowsUnwrappedException() throws Exception {
+        String request = """
+                            {
+                              "jsonrpc": "2.0",
+                              "id": 1,
+                              "method": "tools/call",
+                              "params": {
+                                "name": "excludeExceptionTool",
+                                "arguments": {
+                                  "input": "fail-now"
+                                }
+                              }
+                            }
+                        """;
+
+        String response = HttpTestUtils.callMCP(server, ENDPOINT, request);
+
+        String expectedResponse = """
+                            {
+                              "id": 1,
+                              "jsonrpc": "2.0",
+                              "result": {
+                                "isError": true,
+                                "content": [
+                                  {
+                                    "type": "text",
+                                    "text": "Internal server error: fail-now"
+                                  }
+                                ]
+                              }
+                            }
+                        """;
+
+        JSONAssert.assertEquals(expectedResponse, response, true);
```

Comment:
```suggestion
        String expectedResponse = """
                            {
                              "id": 1,
                              "jsonrpc": "2.0",
                              "result": {
                                "isError": true,
                                "content": [
                                  {
                                    "type": "text",
                                    "text": "Internal server error"
                                  }
                                ]
                              }
                            }
                        """;

        JSONAssert.assertEquals(expectedResponse, response, true);
```

This test should expect a generic "Internal server error" to be returned, since the exception thrown from the tool method does not match any of the exceptions listed in the `@WrapBusinessError` annotation on the tool method.